update_data task for CumulusCI (#3031)
SFDO-Tooling/CumulusCI: docs/data.rst

    cci task run delete_data -o objects Account -o hardDelete True

``update_data``
---------------

> You can update records using CumulusCI. An update consists of two parts:

Reviewer: ```suggestion
To update records using CumulusCI, provide:
```
```suggestion
You can update records using CumulusCI. An update consists of two parts:
```

> * a command line or task configuration saying what to update

Reviewer: ```suggestion
* a command line or task configuration describing what to update
```

* a recipe in a subset of Snowfakery syntax that says how to update it

> Starting from the command line, we can start an update operation like

Reviewer: ```suggestion
On the command line, you can run an update like
```

this:

``$ cci task run update_data --recipe datasets/update.recipe.yml --object Account``

> This will download every Account in the org and apply the fields from the

Reviewer: ```suggestion
This command downloads every Account in the org and applies the fields from the specified
```

update recipe file.

> If we want to filter the rows that we are updating, we can do that too:

Reviewer: ```suggestion
You can filter the rows that you're updating like this:
```

``$ cci task run update_data --recipe datasets/update.recipe.yml --object Account --where "name like 'AAA%'" ``

> The recipe to say what to update can be as simple as this:

Reviewer: ```suggestion
The recipe for an update can be as simple as this:
```

.. code-block::
    - object: Account
      fields:
        NumberOfEmployees: 10000

> But you may also use all of the power of ``snowfakery``, such as fake data:

Reviewer: ```suggestion
You can use all of the power of ``snowfakery`` to add fake data:
```

.. code-block::
    - object: Account
      fields:
        NumberOfEmployees: 10_000
        BillingStreet:
          fake: Streetname

Using Snowfakery formulas, you can also refer to specific input fields
like this:

.. code-block::
    - object: Account
      fields:
        Description: ${{input.Name}} is our favorite customer in ${{input.BillingCity}}

> In order to instruct CumulusCI to extract those fields and make them

Reviewer: ```suggestion
To tell CumulusCI to extract those fields and make them
```

> available to Snowfakery, you should use the ``fields`` option:

Reviewer: ```suggestion
use the ``fields`` option:
```
